cloud/src/autogluon/cloud/predictor/cloud_predictor.py

```diff
@@ -0,0 +1,861 @@
+        train_data,
+        tune_data,
+        config
+    ):
+        cloud_bucket, cloud_key_prefix = s3_path_to_bucket_prefix(self.cloud_output_path)
+        util_key_prefix = cloud_key_prefix + '/utils'
+        train_input = train_data
+        if not is_s3_url(train_data):
+            train_data = self._prepare_data(train_data, 'train')
+            logger.log(20, 'Uploading train data...')
+            train_input = self.sagemaker_session.upload_data(
+                path=train_data,
+                bucket=cloud_bucket,
+                key_prefix=util_key_prefix
+            )
+            logger.log(20, 'Train data uploaded successfully')
+
+        tune_input = tune_data
+        if tune_data is not None:
+            if not is_s3_url(tune_data):
+                tune_data = self._prepare_data(tune_data, 'tune')
+                logger.log(20, 'Uploading tune data...')
+                tune_input = self.sagemaker_session.upload_data(
+                    path=tune_data,
+                    bucket=cloud_bucket,
+                    key_prefix=util_key_prefix
+                )
+                logger.log(20, 'Tune data uploaded successfully')
+
+        config_input = self.sagemaker_session.upload_data(
+            path=config,
+            bucket=cloud_bucket,
+            key_prefix=util_key_prefix
+        )
+        return train_input, tune_input, config_input
+
+    def fit(
+        self,
+        predictor_init_args,
+        predictor_fit_args,
+        leaderboard=True,
+        framework_version='latest',
+        job_name=None,
+        instance_type='ml.m5.2xlarge',
+        instance_count=1,
+        volume_size=30,
+        wait=True,
+        autogluon_sagemaker_estimator_kwargs=dict(),
+        **kwargs
+    ):
+        """
+        Fit the predictor with SageMaker.
+        This function will first upload necessary config and train data to s3 bucket.
+        Then launch a SageMaker training job with the AutoGluon training container.
+
+        Parameters
+        ----------
+        predictor_init_args: dict
+            Init args for the predictor
+        predictor_fit_args: dict
+            Fit args for the predictor
+        leaderboard: bool, default = True
+            Whether to include the leaderboard in the output artifact
+        framework_version: str, default = `latest`
+            Training container version of autogluon.
+            If `latest`, will use the latest available container version.
+            If provided a specific version, will use this version.
+        job_name: str, default = None
+            Name of the launched training job.
+            If None, CloudPredictor will create one with prefix ag-CloudPredictor
+        instance_type: str, default = 'ml.m5.2xlarge'
+            Instance type the predictor will be trained on with SageMaker.
+        instance_count: int, default = 1
+            Number of instance used to fit the predictor.
+        volumes-size: int, default = 30
```

Comment:
what about size for training artifacts? What if models need 100 GB of space?

Reply:
This is actually a good question and I don't have an answer to it. The volume_size is the only parameter related to disk space, and according to the documentation: it is "Size in GB of the EBS volume to use for storing input data during training". Google it doesn't give useful info neither. There are only issues related to not enough disk space for input data, which should be addressed by this parameter